test/subset-band-name/subset_band_name_utitlities.py

```diff
@@ -18,13 +20,35 @@ def get_sds_data(file: str, sds_name: str):
         sds_data = dataset.get()
 
     file_sd.end()
+
     return sds_data
 
 
+def get_all_sds_names(file: str):
+    """Get all SDS names from HDF-4 file"""
+    file_sd = SD(file, SDC.READ)
+    datasets = file_sd.datasets()
+    file_sd.end()
+
+    return datasets
```

Comment:
If you're only using this for names, you should just return the keys from the datasets dictionary. and ignore my comment below. 
```suggestion
    return datasets.keys()
```

Reply:
Going with this approach since it's supposed to return the names.